contrib/epee/include/storages/parserse_base_utils.h

```diff
@@ -30,25 +30,45 @@
 
 #include <algorithm>
 #include <boost/utility/string_ref.hpp>
+#include <boost/optional.hpp>
 
 namespace epee 
 {
 namespace misc_utils
 {
+  class backed_string_ref: public boost::string_ref
```

Comment:
Inheritance should not be used in this situation. Any function that takes `boost::string_ref` by value will "slice away" the "backing" portion of the `std::string`, leaving dangling references. The other "fix" is to ensure every function call uses `boost::string_ref` by const reference which adds minor perf differences due to another pointer indirection. I strongly discourage this "fix"; existing uses of `boost::string_ref` need to be updated and every usage of `boost::string_ref`/`back_string_ref` _from every future commit_ needs to be manually inspected since the compiler will not generate errors with the invalid code.